modules/parent-join/src/main/java/org/elasticsearch/join/mapper/ParentJoinFieldMapper.java

```diff
@@ -85,7 +85,8 @@
     public static ParentJoinFieldMapper getMapper(MapperService service) {
         MetaJoinFieldMapper.MetaJoinFieldType fieldType =
             (MetaJoinFieldMapper.MetaJoinFieldType) service.fieldType(MetaJoinFieldMapper.NAME);
-        return fieldType == null ? null : fieldType.getMapper();
+        return fieldType == null ? null :
+            (ParentJoinFieldMapper) service.fieldMapper(fieldType.getJoinField());
```

Comment:
ParentJoinFieldMapper was using updateFieldType() to update its linked MetaJoinFieldMapper.  This requires a small bit of hackery to get round, which I hope to fix in a followup by moving the various filter factory methods from the field mapper to the field type.